cpp/src/lib/rora_proxy_client.cc

```diff
@@ -221,20 +229,38 @@ int RoraProxy_client::_short_path_many(
   return result;
 }
 
+void _process(std::vector<object_info> &object_infos,
+              const std::string &namespace_) {
+
+  ALBA_LOG(DEBUG, "_process : " << object_infos.size());
+  for (auto &object_info : object_infos) {
+    using alba::stuff::operator<<;
+    // ALBA_LOG(DEBUG, "_procesing object_info:" << object_info);
+
+    const std::string &object_name = std::get<0>(object_info);
+    // auto& future = std::get<1>(object_info);
+    std::unique_ptr<Manifest> mfp = std::move(std::get<2>(object_info));
+
+    _maybe_add_to_manifest_cache(namespace_, object_name, std::move(mfp));
+  }
+}
+
 void RoraProxy_client::read_objects_slices(
     const std::string &namespace_, const std::vector<ObjectSlices> &slices,
     const consistent_read consistent_read_) {
 
   if (consistent_read_ == consistent_read::T) {
-    _delegate->read_objects_slices(namespace_, slices, consistent_read_);
+    std::vector<object_info> object_infos;
+    _delegate->read_objects_slices2(namespace_, slices, consistent_read_,
+                                    object_infos);
+    _process(object_infos, namespace_);
   } else {
     std::vector<ObjectSlices> via_proxy;
     std::vector<short_path_entry> short_path;
     for (auto &object_slices : slices) {
       auto object_name = object_slices.object_name;
-      auto key = strpair(namespace_, object_name);
       auto &cache = ManifestCache::getInstance();
-      auto mfp = cache.find(key);
+      auto mfp = cache.find(namespace_, object_name);
       if (nullptr != mfp &&
           compressor_t::NO_COMPRESSION == mfp->compression->get_compressor() &&
           encryption_t::NO_ENCRYPTION == mfp->encrypt_info->get_encryption()) {
```

Comment:
redundant check as this (no compression, no encryption) was already checked before adding the manifest to the cache